packages/jellyfish-api-core/__tests__/category/account/listPendingFutureSwaps.test.ts

```diff
@@ -0,0 +1,570 @@
+        }
+      })
+
+      it('Should listPendingFutureSwaps if multiple futureswap DUSD for TSLA', async () => {
+        // Call listPendingFutureSwaps before performing futureswap
+        {
+          const pendingFutureSwaps = await testing.rpc.account.listPendingFutureSwaps()
+          expect(pendingFutureSwaps).toStrictEqual([])
+        }
+
+        // futureswap 4 times
+        await testing.container.call('futureswap', [tslaAddress, '0.8@DUSD', 'TSLA'])
+        await testing.generate(1)
+
+        await testing.container.call('futureswap', [tslaAddress, '0.6@DUSD', 'TSLA'])
+        await testing.generate(1)
+
+        await testing.container.call('futureswap', [tslaAddress, '0.4@DUSD', 'TSLA'])
+        await testing.generate(1)
+
+        await testing.container.call('futureswap', [tslaAddress, '0.2@DUSD', 'TSLA'])
+        await testing.generate(1)
+
+        // Call listpendingfutureswaps after performing futureswap
+        const pendingFutureSwaps1 = await testing.rpc.account.listPendingFutureSwaps()
+        expect(pendingFutureSwaps1).toStrictEqual(
+          [
+            {
+              owner: tslaAddress,
+              source: '0.20000000@DUSD',
+              destination: 'TSLA'
+            },
+            {
+              owner: tslaAddress,
+              source: '0.40000000@DUSD',
+              destination: 'TSLA'
+            },
+            {
+              owner: tslaAddress,
+              source: '0.60000000@DUSD',
+              destination: 'TSLA'
+            },
+            {
+              owner: tslaAddress,
+              source: '0.80000000@DUSD',
+              destination: 'TSLA'
+            }
+          ]
+        )
+
+        // Wait for 1 block before getfutureswapblock
+        const nextSettleBlock = await testing.container.call('getfutureswapblock')
+        await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount() - 1)
+
+        const pendingFutureSwaps2 = await testing.rpc.account.listPendingFutureSwaps()
+        expect(pendingFutureSwaps2).toStrictEqual(pendingFutureSwaps1) // Nothing change
+
+        // Generate 1 more block to futureswap
+        await testing.generate(1)
+
+        // Call listpendingfutureswaps after futureswap
+        {
+          const pendingFutureSwaps = await testing.rpc.account.listPendingFutureSwaps()
+          expect(pendingFutureSwaps).toStrictEqual([])
+        }
+      })
+    })
+
+    describe('For multiple tokens which are TSLA, AAPL, MSFT', () => {
+      it('Should listPendingFutureSwaps if multiple futureswaps TSLA, AAPL, MSFT for DUSD', async () => {
+        // Call listpendingfutureswaps before performing futureswap
+        {
+          const pendingFutureSwaps = await testing.rpc.account.listPendingFutureSwaps()
+          expect(pendingFutureSwaps).toStrictEqual([])
+        }
+
+        // futureswap 2 times for every token
+        await testing.container.call('futureswap', [tslaAddress, '0.85@TSLA'])
```

Comment:
I have the tendency to generate a new block after every RPC call. 
The reason I don't do that here is that I don't want to hit the futures swap block